lesson_shell_fasta.md

```diff
@@ -0,0 +1,233 @@
+We will use as sample data set the human [non-coding RNAs](https://en.wikipedia.org/wiki/Non-coding_RNA) (ncRNA).
+
+## Exercises
+### 1 - Downloading data
+
+The data we need are on the [Ensembl](http://www.ensembl.org/index.html) FTP site, a repository of all publicly available files from the project.
+
+There is an [html interface](http://www.ensembl.org/info/data/ftp/index.html) of the last release where it should be easy to locate the link to the FASTA file of the ncRNA
+
+*or*
+ We need to navigate the repository .
+You can navigate the public [FTP](ftp://ftp.ensembl.org/pub/) repository to find the file with all the sequences of human ncRNA (a little help: follow  last\_release>FASTA>homo_sapiens>ncrna)   
+
+*in both cases*
+
+we need to copy the full address that will look like this one:
+
+```
+ftp://ftp.ensembl.org/pub/release-81/fasta/homo_sapiens/ncrna/Homo_sapiens.GRCh38.ncrna.fa.gz
+```
+
+We will use this address as an argument to the command [wget](http://www.gnu.org/software/wget/manual/wget.html#Overview) to retrieve the data. Type in your terminal the command line below to download the file on your local machine.
+
+```
+$ wget <web address of the fasta file>
+```
+
+### 2 - Looking at the data
+
+The file we just downloaded is compressed and in the [gzip](https://www.gnu.org/software/gzip/manual/gzip.html) format. To see how big it is  use the *-l* and *-h* option of *ls*:
+
+```
+$ ls -lh
+
+-rw-rw-r-- 1 usr usr 8,4M ago 13 15:30 Homo_sapiens.GRCh38.ncrna.fa.gz
+```
+
+We can either work with the gzipped file or unzip it before, however because data sets can be really big it is probably worth working with the gzipped file.
+
+We can perform many operations on the compressed file like we do on uncompressed files using "z" commands. We can view the file content using:
+
+```
+$ zcat filename.gz | more
+$ zcat filename.gz | less
+
+```
+or combine them using [pipe](https://en.wikipedia.org/wiki/Pipeline_%28Unix%29)(|):
+
+```
+$ zless filename.gz
+$ zmore filename.gz
+```
+
+For any further analysis we might want to understand how the file is structured and to do so it is probably worth reading a bit of it. Can you identify how a single ncRNA is described?
+
+```
+
+>ENST00000517275 ncrna:known chromosome:GRCh38:2:231757887:231757950:1 gene:ENSG00000253084 gene_biotype:snRNA transcript_biotype:snRNA
+GTGCTTGCTTCGGCAGCACATATACTAAAATTGGAATGATACAGAGAAGATGAGCATGGC
+CCCT
+
+```
+
+It is probably worth downloading and reading the [README companion file](ftp://ftp.ensembl.org/pub/release-81/fasta/homo_sapiens/ncrna/README) located in the same ftp directory of the fasta file we just downloaded. As you will see any ncRNA is bescribed by at least two lines: a header and one (or more) line of DNA sequence.  The README file also contains information on how the header is structured. 
```

Comment:
as you *can* see *each* ncRNA is *described* by at least two

I might restructure these sentences though.

Something like...
As you can see each ncRNA is described by at least two lines.  A header, starting with '>', and at least one line of DNA sequence.  To learn what information is included in the header, we can download the README companion file located in the same ftp directory as the fasta file we just downloaded.